src/App.js

```diff
@@ -0,0 +1,272 @@
+  LogoutOutlined
+} from '@ant-design/icons';
+
+const { Sider, Content } = Layout;
+const { Text } = Typography;
+const authService = new AuthService();
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      transactions: [],
+      aggregatedTransactions: [],
+      collapsed: false,
+      redirect: false,
+      loggedIn: authService.isLoggedIn(),
+      pageTitle: 'My Transactions'
+    }
+  }
+
+  setData = async (token, del = false) => {
+    try {
+      const config = {
+        headers: { Authorization: `Bearer ${token}` }
+      };
+      const resp = await axios.get('/transactions', config);
+      const transactions = resp.data.data;
+      const catNames = this.getCategoryNames(transactions);
+      const aggregatedTransactions = await this.getAggrData(catNames);
+      if (del) { return this.setState({ aggregatedTransactions, transactions }, message.success('Transaction deleted', 5)) }
+      this.setState({ aggregatedTransactions, transactions });
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  getTransactionsData = async () => {
+    try {
+      const token = authService.getToken()
+      const config = {
+        headers: { Authorization: `Bearer ${token}` }
+      };
+
+      const resp = await axios.get('/transactions', config);
+      const transactions = resp.data.data;
+      const catNames = this.getCategoryNames(transactions);
+      const aggregatedTransactions = await this.getAggrData(catNames);
+
+      return { transactions, aggregatedTransactions }
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  getAggrData = async (categoryNames) => {
+    const aggrTransactions = []
+    categoryNames.forEach(async (cat) => {
+      const aggr = await this.getAggrCategory(cat);
+      aggrTransactions.push(aggr);
+    })
+    return aggrTransactions;
+  }
+
+  componentDidMount = async () => {
+    if (authService.isLoggedIn()) {
+      try {
+        const { transactions, aggregatedTransactions } = await this.getTransactionsData();
+        this.setState({ transactions, aggregatedTransactions });
+      } catch (error) {
+        throw error;
+      }
+    }
+  }
+
+  getCategoryNames = (transactions) => {
+    const names = {}
+    transactions.forEach(t => {
+      if (!names[t.category]) {
+        names[t.category] = true;
+      }
+    })
+    return Object.keys(names)
+  }
+
+  getAggrCategory = async (category) => {
+    if (authService.isLoggedIn()) {
+      const token = authService.getToken()
+      try {
+        const config = {
+          headers: { Authorization: `Bearer ${token}` }
+        };
+        const resp = await axios.get(`transactions/${category}`, config);
+        const aggrCat = {}
+        aggrCat[category] = resp.data.data[0].amount;
+        return aggrCat
+      } catch (error) {
+        throw error;
+      }
+    }
+  }
+
+  renderBalance = () => {
+    let balance = 0;
+    this.state.transactions.forEach(t => balance += t.amount)
```

Comment:
check [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) method...

Reply:
actually started doing it with reduce and got a little confused.
I will retry.